test/rpc_test.py

```diff
@@ -533,6 +533,28 @@ def test_py_user_defined(self):
         )
         self.assertEqual(ret, my_function(n, n + 1, n + 2))
 
+    def _profiler_test_with_rpc(self, use_async=False):
+        dst = (self.rank + 1) % self.world_size
+        # only run profiler on rank 0.
+        if self.rank in {0}:
+            with torch.autograd.profiler.profile() as prof:
+                if not use_async:
+                    rpc.rpc_sync("worker{}".format(dst), my_sleep_func, args=(1, ))
+                else:
+                    fut = rpc.rpc_async("worker{}".format(dst), my_sleep_func, args=(1, ))
+                    time.sleep(3)
```

Comment:
This is to test that even though the whole function itself takes ~4 seconds, the async_rpc is profiled as only having taken ~1 second. We should have this because, for instance, without the changes to the threading logic in the profiler, this will incorrectly report 4 seconds for the async_rpc.